### PR TITLE
Create workspace renaming endpoint

### LINKED ISSUE #39: Add support for renaming workspaces
Currently this does not exist, but it existed with the previous API.

### PR DESCRIPTION
**Pull request by eagw**
Creates a new endpoint that re-enables workspace renaming functionality by making a new WorkspaceRenameSerializer and using it in a new update() function in the WorkspaceViewset class.

Works in conjunction with the rename-workspace branches of [multinetjs](https://github.com/multinet-app/multinetjs/tree/rename-workspace) and [multinet-client](https://github.com/multinet-app/multinet-client/tree/rename-workspace).

Resolves #39

### CODE REVIEW

File: multinet/api/views/workspace.py
Comment (waxlamp) [inline multinet/api/views/workspace.py:56]: According to the [permissions RFC](https://github.com/multinet-app/multinet-rfcs/tree/master/0005-permissions_model#roles), this should be a `maintainer` level action. @AlmightyYakob do I have that right?

**Review by waxlamp — CHANGES_REQUESTED** [review]

File: multinet/api/views/workspace.py
Comment (waxlamp) [inline multinet/api/views/workspace.py:56]: A general question for @AlmightyYakob and @naglepuff--I'm surprised to see stringly-typed role names here, rather than symbols defined in an enum. Did I miss the boat on a newer way of doing things?

Comment (naglepuff) [reply multinet/api/views/workspace.py:56]: Not sure when this branch was created but it is behind master. I don't think it has any of the new permissions stuff in it. I'm not a git expert but I think a rebase would be good here.

Comment (waxlamp) [reply multinet/api/views/workspace.py:56]: A merge in this case would be better than a rebase (since reviewing has already begun).

@eagw, when you do merge master into your branch, you may run into merge conflicts. If you do, feel free to ask me or Jake for help in resolving them.

Comment (eagw) [reply multinet/api/views/workspace.py]: It's now merged and the new permissions model has been used.

See 78d4b3f and 3fbfbc3

**Review by waxlamp — CHANGES_REQUESTED** [review]
Could you run the formatter on your codebase? That should resolve the linting errors causing the tests to fail. The command should be

```
docker-compose run --rm django tox -e format
```

This will reformat your code to be compliant with the linting rules. You should commit any changed files with a message about fixing linting errors.

**Review by waxlamp — COMMENTED** [review]
Looks good to me, but please defer to @AlmightyYakob's review.

**Comment by AlmightyYakob** [comment]
I think we should add a test here, to both test renaming functionality as well as permission handling. @eagw Lmk if you want some guidance here.

**Comment by eagw** [comment]
> I think we should add a test here, to both test renaming functionality as well as permission handling. @eagw Lmk if you want some guidance here.

@AlmightyYakob How does 25d5a22 look for a test?

File: multinet/api/tests/test_workspace.py
Comment (AlmightyYakob) [inline multinet/api/tests/test_workspace.py:143]: Small style suggestion but otherwise I think you nailed it!
```suggestion
    new_name = Faker().pystr()

    r = authenticated_api_client.put(
        f'/api/workspaces/{workspace.name}/',
        {
            'name': new_name,
        },
        format='json',
    )
    assert r.status_code == status_code

    # Retrieve workspace to ensure it's up to date
    workspace = Workspace.objects.get(id=workspace.pk)

    # Assert name is as expected
    expected_name = new_name if success else old_name
    assert workspace.name == expected_name
```

**Review by AlmightyYakob — APPROVED** [review]
Looks great, thanks @eagw!

**Pull request merged**
